src/systolic_array/commit.cpp

```diff
@@ -186,9 +190,10 @@ void Commit::localSpadCallback(PacketPtr pkt) {
 
 void Commit::queueCommitRequest(int lineIndex) {
   Addr addr = iter * accel.elemSize;
-  uint8_t* data = new uint8_t[accel.lineSize]();
+  int reqSize = std::min<int>(elemsPerLine, inputs.size()) * accel.elemSize;
+  uint8_t* data = new uint8_t[reqSize];
   // Copy data from the buffer for the collected data.
-  for (int i = 0; i < elemsPerLine; i++) {
+  for (int i = 0; i < reqSize / accel.elemSize; i++) {
```

Comment:
In general, integer divides make me nervous, because they're so easy to get wrong in C++ and the compiler often won't even warn you.  In general, I would extract divisions into temporary variables so that any bugs can only happen in one place. In this case, the divide is actually unnecessary, because this is the same as `min(elemsPerLine, inputs.size())`.

Reply:
I agree. I changed this function to take two arguments: the start index of the elements, and the number of elements to write. This is clearer to me.